lib/components/ModalWrapper/index.tsx

```diff
@@ -43,31 +44,32 @@ class ModalWrapperImpl extends React.PureComponent<Props, {}> {
    using Portal kinda disturbs ClickOutside.
    let's do some logic here to check that the click outside
    is definitely outside ModalWrapper before calling hide
+   If no onClickOutside callback sent use onCancel
    **/
   handleClickOutside(e: React.SyntheticEvent<HTMLElement>) {
-    if (!this.modalWrapper) return;
-
     const target = e.target as HTMLElement;
-    if (e.currentTarget === target && this.props.onCancel) this.props.onCancel(e);
+    const isFinalTarget = e.currentTarget === target;
+    if (!isFinalTarget || !this.modalWrapper || !this.props.onClickOutside) return;
+
+    this.props.onClickOutside(e);
   }
 
   setModalWrapperRef(modalWrapper: HTMLDivElement | null) {
     this.modalWrapper = modalWrapper;
   }
 
   render() {
-    const { classes, onCancel, children } = this.props;
+    const { classes, onClickOutside, children } = this.props;
 
     return (
       <div className={classes.container} ref={this.setModalWrapperRef} onClick={this.handleClickOutside}>
-        { onCancel &&
+        { onClickOutside &&
         <KeyHandler
```

Comment:
I don't think it's ok to bind a `onClickOutside` on a keyboard keyhandler ("esc" here).